File: app/src/main/java/com/example/notificationapp/data/adapters/ClubListAdapter.java
package com.example.notificationapp.data.adapters;


import android.content.Context;
import android.content.Intent;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatTextView;
import androidx.recyclerview.widget.RecyclerView;

import com.example.notificationapp.R;
import com.example.notificationapp.data.network.model.ClubModel;
import com.example.notificationapp.utils.Constants;
import com.example.notificationapp.view.activities.ClubActivity;

import java.util.List;

public class ClubListAdapter extends RecyclerView.Adapter<ClubListAdapter.CustomVH> {

    private final Context mContext;
    private final List<ClubModel> mClubs;

    public ClubListAdapter(List<ClubModel> clubs, Context context) {
        this.mClubs = clubs;
        this.mContext = context;
    }

    @NonNull
    @Override
    public CustomVH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new CustomVH(View.inflate(parent.getContext(), R.layout.club_list_item, null));
    }

    @Override
    public void onBindViewHolder(@NonNull CustomVH holder, int position) {
        holder.setView(mClubs.get(position));
    }

    @Override
    public int getItemCount() {
        return mClubs.size();
    }

    public class CustomVH extends RecyclerView.ViewHolder {

        private final AppCompatTextView name, description;

        public CustomVH(@NonNull View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.club_name);
            description = itemView.findViewById(R.id.textDescription);
            itemView.setOnClickListener(v -> {
                Intent intent = new Intent(mContext, ClubActivity.class);
                intent.putExtra(Constants.CLUB_NAME, mClubs.get(getAdapterPosition()).getName());
                intent.putExtra(Constants.CLUB_ID, mClubs.get(getAdapterPosition()).getId());
                intent.putExtra(Constants.CLUB_DESC, mClubs.get(getAdapterPosition()).getDescription());
                mContext.startActivity(intent);
            });
        }

        public void setView(ClubModel clubModel) {
            name.setText(clubModel.getName());
            description.setText(clubModel.getDescription());
        }
    }
}

File: app/src/main/java/com/example/notificationapp/data/adapters/ClubListAdapter.kt
package com.example.notificationapp.data.adapters

import android.content.Context
import android.content.Intent
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatTextView
import androidx.recyclerview.widget.RecyclerView
import com.example.notificationapp.Constants
import com.example.notificationapp.R
import com.example.notificationapp.data.adapters.ClubListAdapter.CustomVH
import com.example.notificationapp.data.network.ClubModel
import com.example.notificationapp.view.activities.ClubActivity
import com.google.android.material.card.MaterialCardView

class ClubListAdapter(private val mClubs: List<ClubModel>, private val mContext: Context) : RecyclerView.Adapter<CustomVH>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
        CustomVH(View.inflate(parent.context, R.layout.club_list_item, null))

    override fun onBindViewHolder(holder: CustomVH, position: Int) {
        holder.bindView(mClubs[position])
    }

    override fun getItemCount(): Int = mClubs.size

    inner class CustomVH(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private val name: AppCompatTextView = itemView.findViewById(R.id.club_name)
        private val description: AppCompatTextView = itemView.findViewById(R.id.textDescription)
        private val background: MaterialCardView = itemView.findViewById(R.id.item_background)

        fun bindView(clubModel: ClubModel) {
            name.text = clubModel.name
            description.text = clubModel.description
            background.setOnClickListener {
                val intent = Intent(mContext, ClubActivity::class.java)
                intent.putExtra(Constants.CLUB_NAME, mClubs[adapterPosition].name)
                intent.putExtra(Constants.CLUB_ID, mClubs[adapterPosition].id)
                intent.putExtra(Constants.CLUB_DESC, mClubs[adapterPosition].description)
                mContext.startActivity(intent)
            }
        }
    }
}